apps/services/user-notification/src/app/modules/notifications/notifications.service.ts

```diff
@@ -55,69 +48,37 @@ export class NotificationsService {
   constructor(
     @Inject(LOGGER_PROVIDER)
     private readonly logger: Logger,
-    @Inject(CACHE_MANAGER) private cacheManager: Cache,
     @InjectModel(Notification)
     private readonly notificationModel: typeof Notification,
+    private readonly contentfulGraphQLClientService: ContentfulGraphQLClientService,
+    
   ) {}
 
-  async performGraphQLRequest(query: string) {
-    try {
-      const response = await axios.post(
-        CONTENTFUL_GQL_ENDPOINT,
-        { query },
-        {
-          headers: {
-            'content-type': 'application/json',
-            authorization: `Bearer ${ACCESS_TOKEN}`,
-          },
-        },
-      )
-      return response.data
-    } catch (error) {
-      if (error.response) {
-        throw new BadRequestException(error.response.data)
-      } else {
-        this.logger.error('GraphQL Request Failed', { error })
-        throw new InternalServerErrorException('Internal Server Error')
-      }
-    }
+  private cleanStringAdvanced(str:string) {
+    return str.trim() // Remove leading and trailing whitespace
+              .replace(/\t/g, '') // Remove tabs
+              .replace(/\n/g, '') // Remove newlines
   }
-
-  async getSenderOrganization(
+  async getSenderOrganizationTitle(
     senderId: string,
     locale?: Locale,
-  ): Promise<SenderOrganization> {
+  ): Promise<SenderOrganization | undefined> {
     locale = mapToLocale(locale as Locale)
-    const cacheKey = `org-${senderId}-${locale}`
-    const cachedOrganization = await this.cacheManager.get<SenderOrganization>(
-      cacheKey,
-    )
-    if (cachedOrganization) {
-      this.logger.info(`Cache HIT for: ${cacheKey}`, cachedOrganization)
-      return cachedOrganization
+    const queryVariables = {
+      kennitala: senderId,
+      locale: mapToContentfulLocale(locale),
+    };
+    const res = await this.contentfulGraphQLClientService.fetchData(
+      GetOrganizationByKennitala, 
```

Comment:
When looking at this, I'm wondering if we should have just added a `organization.service.ts` in the `libs/cms` library, with a function `getOrganizationByNationalId` 🤔

Then similar for hnipp templates `hnipp-template.service.ts` with function `getTemplates` and `getTemplateById`.

I maybe should have asked, why do we want to use the gql endpoint over the sdk client? Have we benchmarked it if it is for performance reasons?

Reply:
hmmm ... i think that is a bigger discussion where we should compare the two in a variety of dimensions
but in short benchmark no

Reply:
i would opt to use the gql endpoint over the sdk just for simplicity and less overhead in code